src/services/api/routes/communities.ts

```diff
@@ -284,4 +302,122 @@ router.post('/', multer().none(), async function (request: express.Request, resp
 	}).end({ pretty: true, allowEmpty: true }));
 });
 
+router.post('/:community_action', multer().none(), async function (request: express.Request, response: express.Response): Promise<void> {
+	response.type('application/xml');
+
+	const parentCommunity: HydratedCommunityDocument | null = await getCommunityByTitleID(request.paramPack.title_id);
+	if (!parentCommunity) {
+		await respondCommunityNotFound(response);
+		return;
+	}
+
+	const communityActions: string[] = request.params.community_action.split('.');
+	const community: HydratedCommunityDocument | null = await commonGetSubCommunity(request, response, communityActions[0]);
+	if (!community) {
+		return;
+	}
+
+	if (communityActions.length > 1) {
+		if (communityActions[1] === 'delete') {
+			await deleteSubCommunity(community, request, response);
+		} else if (communityActions[1] === 'favorite') {
+			await addFavoriteSubCommunity(community, request, response);
+		} else if (communityActions[1] === 'unfavorite') {
+			await delFavoriteSubCommunity(community, request, response);
+		} else { // '%s.search'
+			response.sendStatus(501); // Not Implemented
+		}
+	} else {
+		await updateSubCommunity(community, request, response);
+	}
+});
+
+async function updateSubCommunity(community: HydratedCommunityDocument, request: express.Request, response: express.Response): Promise<void> {
+	response.type('application/xml');
+
+	if (community.owner != request.pid) {
+		response.sendStatus(403); // Forbidden
+		return;
+	}
+
+	const bodyCheck: z.SafeParseReturnType<CreateNewCommunityBody, CreateNewCommunityBody> = createNewCommunitySchema.safeParse(request.body);
+	if (!bodyCheck.success) {
+		response.send(xmlbuilder.create({
+			result: {
+				has_error: 1,
+				version: 1,
+				code: 404,
+				error_code: 20,
+				message: 'BAD_COMMUNITY_DATA'
+			}
+		}).end({ pretty: true }));
+		return;
+	}
+
+	community.name = request.body.name;
+	community.description = request.body.description;
+	community.icon = request.body.icon;
+	community.app_data = request.body.app_data;
+	await community.save();
+
+	response.send(xmlbuilder.create({
+		result: {
+			has_error: '0',
+			version: '1',
+			request_name: 'community',
+			community: community.json()
+		}
+	}).end({ pretty: true, allowEmpty: true }));
+}
+
+async function deleteSubCommunity(community: HydratedCommunityDocument, request: express.Request, response: express.Response): Promise<void> {
+	response.type('application/xml');
+
+	if (community.owner != request.pid) {
+		response.sendStatus(403); // Forbidden
+		return;
+	}
+
+	await Community.deleteOne({ _id: community._id });
```

Comment:
For analytics purposes it would be nice to actually keep these documents around, unless that would cause issues of some kind. This is used in MK8 for tournaments correct? It would be nice to have this data around to reference to be able to tell users like "You started X tournaments with Y unique players", or let people download the data from their past tournaments or something (I have not actually looked into the data these subcommunities store). Maybe instead of deleting it outright we could mark communities as "active" or something? For PNIDs I just have a `deleted` flag on them

Though maybe that's better left for a future PR, what do you think? I'm fine with waiting I just figure that instead of going back and doing this later we might as well get that ball rolling while already messing with community stuff

Reply:
I think we can do that in another PR.

Also the stats are probably better to do from the MK8 server itself, since a SimpleSearchObject (tournament) is tied to a CommunityID and Code, we can know who joins it, who has a ranking in it etc..

Reply:
Thats fair., sounds good We can leave it as is for now, and look more into going forward to see what the best path would be. I don't collect any analytics anywhere else anyway so it's not like this is dire